In [1]:
import pandas as pd
import matplotlib.pyplot as plt
import numpy as np

In [2]:
df=pd.read_csv("cookie_cats.csv")
df.head()

userid  version  sum_gamerounds  retention_1  retention_7
0     116  gate_30               3        False        False
1     337  gate_30              38         True        False
2     377  gate_40             165         True        False
3     483  gate_40               1        False        False
4     488  gate_40             179         True         True

In [3]:
df.dtypes

userid             int64
version           object
sum_gamerounds     int64
retention_1         bool
retention_7         bool
dtype: object

## Lets first check if the number of users in both the cases are same

In [62]:
count_each=df.groupby("version")["userid"].count().reset_index()
count_each

version  userid
0  gate_30   44700
1  gate_40   45489

## Almost same number of user. Thats good. 
## Now we check, what is the average number of games played by a user of a version.

In [5]:
avg_games=df.groupby(["version"])["sum_gamerounds"].mean()
avg_games

version
gate_30    52.456264
gate_40    51.298776
Name: sum_gamerounds, dtype: float64

## Again same number of average games played.
## Average is almost same. Lets see if the data is also uniform or it has some outliers.

In [44]:
hist=df.loc[df["version"]=="gate_30"][["sum_gamerounds","userid"]]
hist=hist.sort_values(by="sum_gamerounds",ascending=False).reset_index()
hist["sum_gamerounds"].describe()


count    44700.000000
mean        52.456264
std        256.716423
min          0.000000
25%          5.000000
50%         17.000000
75%         50.000000
max      49854.000000
Name: sum_gamerounds, dtype: float64

In [31]:
hist2=df.loc[df["version"]=="gate_40"][["sum_gamerounds","userid"]]
hist2=hist2.sort_values(by="sum_gamerounds",ascending=False).reset_index()
hist2["sum_gamerounds"].describe()


count    45489.000000
mean        51.298776
std        103.294416
min          0.000000
25%          5.000000
50%         16.000000
75%         52.000000
max       2640.000000
Name: sum_gamerounds, dtype: float64

 ## As we can see gate_30 has more standard deviation than gate_40. Both perform apporximately equal till 75 percentile. One user has played the game 49854 amount of time which is causing a huge difference in standard deviation.

In [47]:
Count_0=df[df["sum_gamerounds"]==0].groupby("version").count()
Count_0


userid  sum_gamerounds  retention_1  retention_7
version                                                  
gate_30    1937            1937         1937         1937
gate_40    2057            2057         2057         2057

In [ ]:
## Almost 2000 users in both the games didnt play a single game and uninstalled.

In [52]:
total_0 = df[df["sum_gamerounds"]==0]["userid"].count()
total = df["userid"].count()
worst= total_0*100/total
worst

4.42847797403231

In [ ]:
## 4% of total users uninstalled without playing a game.

In [68]:
count_retention_1 = df[df["retention_1"]==True].groupby("version")["userid"].count().reset_index()
merge=pd.merge(count_each,count_retention_1,on="version")
merge
count_retention_7 = df[df["retention_7"]==True].groupby("version")["userid"].count().reset_index()
merge = pd.merge(merge,count_retention_7,on="version")
merge.columns=["version","Total_users","1day_retention","7day_retention"]
merge["1day_retention%"]=merge["1day_retention"]*100/merge["Total_users"]
merge["7day_retention%"]=merge["7day_retention"]*100/merge["Total_users"]
merge

version  Total_users  1day_retention  7day_retention  1day_retention%  \
0  gate_30        44700           20034            8502        44.818792   
1  gate_40        45489           20119            8279        44.228275   

   7day_retention%  
0        19.020134  
1        18.200004

In [71]:
## This shows that gate_30 version performs better in three terms 
## 1. 1% more retention 
## 2. Every user plays one extra round of game.